### PR TITLE
Ping/113 azure oauth

### PR DESCRIPTION
**Pull request by pingchenchan**
## Description

1. OAuth with Azure Entra ID:
A Azure tenant was created using my personal account for Azure Entra ID to authenticate users via OAuth.
2. Role-Based Redirection
Roles have been defined for users (Admin and Volunteer). Upon successful login: Admin users are redirected to the Inventory page.Volunteer users are redirected to the Pick Your Name page.
3. Updated Styling:
The UI has been updated to align with the latest Figma design
4. Testing the Application:
You can test the application with the following dummy accounts:
  - Admin Account:
    - Email: pingchenchan@outlook.com
    - Password: digitalaidseattle123***
  - Volunteer Account:
    - Email: pingchenchan@gmail.com
    - Password: digitalaidseattle123***

## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Related Tickets & Documents

[PIT113 ](https://das-ph-inventory-tracker.atlassian.net/jira/core/projects/PIT/board?selectedIssue=PIT-113)

_Link user story from projects.digitalaidseattle.org_

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

1. FYI, this is how I create a tenant and then set auth in our local app

https://learn.microsoft.com/en-us/entra/external-id/customers/tutorial-single-page-app-react-sign-in-prepare-tenant

2. This is how to create role and set roles for users
- Create app role in  App registrations/tenant/App roles/create app role
image
image

-  Assigning app roles: App roles for Users/Groups. Assign app roles with 'User' allowed member types in Enterprise applications or in Microsoft Graph APIs.
image

- Redirect users based on their role when they log into the application
image

### CODE REVIEW

File: src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx [hunk 1]
```diff
@@ -75,22 +74,20 @@ const Profile = () => {
   const theme = useTheme();
   const { user } = useContext(UserContext);
   const [username, setUsername] = useState<string>('');
-  const [avatar, setAvatar] = useState<string>('');
-  const navigate = useNavigate();
+  // const [avatar, setAvatar] = useState<string>('');
```
Comment (bospoort) [inline src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx:77]: if there is commented out code, can you either remove or add //TODO or additional comments why we still need it?

File: src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx [hunk 2]
```diff
@@ -75,22 +74,20 @@ const Profile = () => {
...
 
   useEffect(() => {
     if (user) {
-      setAvatar(user.user_metadata.avatar_url);
-      setUsername(
-        user.user_metadata.name
-          ? user.user_metadata.name
-          : user.user_metadata.email,
-      );
+      // setAvatar(user.user_metadata.avatar_url)
```
Comment (bospoort) [inline src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx:82]: same here for the comments

File: src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx [hunk 3]
```diff
@@ -135,7 +132,7 @@ const Profile = () => {
         <Stack direction="row" spacing={2} alignItems="center" sx={{ p: 0.5 }}>
           <Avatar
             alt="profile user"
-            src={avatar}
+            // src={avatar}
```
Comment (bospoort) [inline src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx:135]: comment :-)

File: src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx [hunk 4]
```diff
@@ -189,7 +186,7 @@ const Profile = () => {
                           >
                             <Avatar
                               alt="profile user"
-                              src={avatar}
+                              // src={avatar}
```
Comment (bospoort) [inline src/layout/MainLayout/Header/HeaderContent/Profile/index.tsx:189]: comment :-)

File: src/layout/MainLayout/index.tsx [hunk 5]
```diff
@@ -17,31 +17,42 @@ import Drawer from './Drawer';
...
+        })
+        .then((response) => {
+          const userClaims = response.idTokenClaims;
+          setUser(userClaims || null);
+        })
+        .catch((error) => {
+          console.error('Error acquiring token', error);
+        });
+    } else {
+      console.log('no account navigate("/login"); ');
```
Comment (bospoort) [inline src/layout/MainLayout/index.tsx:53]: Can you be more explicit what the "no account navigate login" means? Is it "No account. Navigate back to /login"? 
(If I see that in console output, I know what is going on)

File: src/pages/authentication/auth-forms/FirebaseSocial.tsx [hunk 6]
```diff
@@ -4,21 +4,38 @@ import { useTheme } from '@mui/material/styles';
...
+    instance
+      .loginPopup(loginRequest)
+      .then((response) => {
+        instance.setActiveAccount(response.account);
+        const idTokenClaims = response.idTokenClaims as { roles?: string[] };
+        const roles = idTokenClaims.roles || [];
+
+        if (roles.includes('admin')) {
+          navigate('/inventory');
+        } else if (roles.includes('volunteer')) {
```
Comment (bospoort) [inline src/pages/authentication/auth-forms/FirebaseSocial.tsx:30]: Beautiful!!!

File: src/pages/authentication/auth-forms/FirebaseSocial.tsx
Comment (bospoort) [inline src/pages/authentication/auth-forms/FirebaseSocial.tsx:1]: Can you change the name to AzureAuth or something

File: src/sections/tickets/TicketDialog.tsx [hunk 7]
```diff
@@ -24,7 +24,7 @@ import {
 } from '@mui/material';
 import { Ticket, ticketService } from './ticketService';
 import useAppConstants from '../../services/useAppConstants';
-import { UserContext } from '../../components/contexts/UserContext';
+// import { UserContext } from '../../components/contexts/UserContext';
```
Comment (bospoort) [inline src/sections/tickets/TicketDialog.tsx:27]: Lots of commented out code. Please add TODO before we leave it in.

File: src/services/authService.ts [hunk 8]
```diff
@@ -5,26 +5,41 @@
...
+  // AuthError,
   OAuthResponse,
   User,
-  UserResponse,
+  //  UserResponse,
 } from '@supabase/supabase-js';
 import { supabaseClient } from './supabaseClient';
 
+//////////////
+const user: User = {
```
Comment (bospoort) [inline src/services/authService.ts:16]: Do we still need all this? It was a quick workaround, but I don't think we use it anymore now that you fixed it.

**Review by bospoort — CHANGES_REQUESTED** [review]
Stunning work. Big PR, that is unfortunate but hard to avoid at this stage. We'll get better at smaller ones.

File: src/services/authService.ts (hunk 8 above)
Comment (pingchenchan) [reply src/services/authService.ts:16]: Yes, we can remove this code

File: src/pages/authentication/auth-forms/AzureAuth.tsx [hunk 9]
```diff
@@ -4,21 +4,38 @@ import { useTheme } from '@mui/material/styles';
...
-import { authService } from '../../../services/authService';
-import { loggingService } from '../../../services/loggingService';
 
+//azure oauth
+import { useMsal } from '@azure/msal-react';
+import { loginRequest } from '../../../authConfig'; // see authConfig.ts code bellow
+import { useNavigate } from 'react-router-dom';
 // ==============================|| FIREBASE - SOCIAL BUTTON ||============================== //
 
 const FirebaseSocial = () => {
```
Comment (bospoort) [inline src/pages/authentication/auth-forms/AzureAuth.tsx:14]:
```suggestion
const AzureAuth = () => {
```

**Review by bospoort — APPROVED** [review]
Just one last rename and I'm good!!

**Review by aitzeng — APPROVED** [review]

**Pull request merged**